### PR TITLE
Adds rotation matrix as method of Matrix3.

### PR DESCRIPTION
**Pull request by g-bauer**
This PR moves the function `rotation_matrix` from the rotational MC move to be a method of `Matrix3` and adds a doctest.

I also modified the old function to compute the norm of `axis` inside the function.

### CODE REVIEW

**Comment by Luthaf** [comment]
They may have been a misunderstanding: the `Cargo.toml` file should stay around. The `Cargo.lock` is the one that should not be checked in the repo =)

**Comment by g-bauer** [comment]
Oops! That was not a missunderstanding, I simply removed the wrong file on accident.

File: src/core/src/types/matrix.rs [hunk 1]
```diff
@@ -91,6 +91,51 @@ impl Matrix3 {
...
+    /// let angle = 90f64.to_radians();
+    ///
+    /// let rotation = Matrix3::rotation_matrix(&(e1*2.0), angle);
+    /// let mut rot_vec = rotation * e3;
+    /// for i in 0..3 {
+    ///     rot_vec[i] = (rot_vec[i] * 1.0e8).round() / 1.0e8
+    /// };
+    /// assert_eq!(rot_vec, Vector3D::new(0.0, 1.0, 0.0));
+    /// ```
+    pub fn rotation_matrix(axis: &Vector3D, angle: f64) -> Matrix3 {
```
Comment (Luthaf) [inline src/core/src/types/matrix.rs]: Should we call this function `rotation`? The matrix part is always implied now that this is a method

Comment (g-bauer) [inline src/core/src/types/matrix.rs]: Good idea!

**Comment by Luthaf** [comment]
Can you squash the commits? I'll merge after that!

**Comment by g-bauer** [comment]
Done. Thank you!

**Comment by Luthaf** [comment]
Thank you !

**Pull request merged**